packages/components/src/font-size-picker/test/index.tsx

```diff
@@ -525,6 +525,39 @@ describe( 'FontSizePicker', () => {
 			expect( screen.getByLabelText( 'Custom' ) ).toBeInTheDocument();
 		} );
 
+		it( 'hides custom input when disableCustomFontSizes is set to `true` with a custom font size', () => {
+			const { rerender } = render(
+				<FontSizePicker fontSizes={ fontSizes } value="3px" />
+			);
+			expect( screen.getByLabelText( 'Custom' ) ).toBeInTheDocument();
+
+			rerender(
+				<FontSizePicker
+					disableCustomFontSizes
+					fontSizes={ fontSizes }
+					value="3px"
+				/>
+			);
+			expect(
+				screen.queryByLabelText( 'Custom' )
+			).not.toBeInTheDocument();
```

Comment:
I didn't write a more specific assertion on which UI is presented to the user for two reasons since I initially wrote this unit test so that it would pass on `trunk`, where no input UI is actually presented to the user in this case (a bug IMO — see a more extended explanation in the PR description).


Since the input UI presented to the user changes depending on the length of the `fontSizes` prop, if we think that we should test that part too, I will duplicate this test and add it to two different sections of the file